libpod/container_exec.go

```diff
@@ -775,12 +776,30 @@ func (c *Container) Exec(config *ExecConfig, streams *define.AttachStreams, resi
 }
 
 // cleanup an exec session after its done
-func (c *Container) cleanupExecBundle(sessionID string) error {
-	if err := os.RemoveAll(c.execBundlePath(sessionID)); err != nil && !os.IsNotExist(err) {
-		return err
+func (c *Container) cleanupExecBundle(sessionID string) (Err error) {
+	path := c.execBundlePath(sessionID)
+	for attempts := 0; attempts < 50; attempts++ {
+		Err = os.RemoveAll(path)
+		if Err == nil || os.IsNotExist(Err) {
+			return nil
+		}
+		if pathErr, ok := Err.(*os.PathError); ok {
+			Err = pathErr.Err
+			if errors.Cause(Err) == unix.ENOTEMPTY || errors.Cause(Err) == unix.EBUSY {
+				// give other processes a chance to use the container
+				if !c.batched {
+					c.lock.Unlock()
+				}
+				time.Sleep(time.Millisecond * 100)
+				if !c.batched {
```

Comment:
Is there any special reason for a `Lock` we continue the loop from here so if another process performs a `rm` we will get `ENOENT` or `os.IsNotExist(`

Reply:
Should we just return if Err not  unix.ENOTEMPTY or unix.EBUSY

Reply:
@flouthoc `ENOENT` must be considered success.  We want to `rm -rf` that directory so if someone else took care of it, that is good too :-)  in fact, it could have been the podman cleanup process to do it.

@rhatdan if Err not unix.ENOTEMPTY or unix.EBUSY, we just return since we don't hit the `continue`

Reply:
Your in a `for` loop and will attempt to remove 50 times before returning is the way I read it?

Reply:
Never mind, I see the return now.